service/history/workflow/context.go

```diff
@@ -179,8 +179,11 @@ func NewContext(
 	metricsHandler metrics.Handler,
 ) *ContextImpl {
 	return &ContextImpl{
-		workflowKey:     workflowKey,
-		logger:          logger,
+		workflowKey: workflowKey,
+		logger: log.With(logger,
+			tag.WorkflowNamespaceID(workflowKey.NamespaceID),
+			tag.WorkflowID(workflowKey.WorkflowID),
+			tag.WorkflowRunID(workflowKey.RunID)),
```

Comment:
This logger is passed into the new Update registry here: https://github.com/temporalio/temporal/pull/6299/files#diff-c365166a3b00b43950b46bc9be819b9c407f41947377a563c8b18369bd5d897fR911